### ISSUE TITLE
Don't ask for interaction if piping to another command

### ISSUE THREAD

**Issue by stahnma** [flox/flox#556]
**Describe the bug:**

flox is waiting for input when I'm trying to pipe the output to another command. 

**Steps to reproduce:**
1. Have an environment in `$HOME`
2. Have another one in another directory
3. Activate the one in `$HOME`
4. `cd` to other directory (without activating that env)
5. run `flox list` -- notice that it asks some questions
6. run something like `flox list | xargs` -- notice it still asks questions. 

Result: 

It hangs waiting on input from user. 
Expected: 
It doesn't do that in a piped scenario. It just picks a behavior ... either always the active env or always the one in `$PWD`. 

Flox version:

Version: 0.3.7-r790

**Screenshots/terminal output (if relevant):**

**Comment by mkenigs** [comment]
We're checking stdin and stderr but not stdout:
https://github.com/flox/flox/blob/3766d1a1e2bf783a1f9062edad17cc3bf27382e3/crates/flox/src/utils/dialog.rs#L126

Is it common to expect stdout should be a tty? I did find this only mentioning stdin: https://clig.dev/#interactivity

**Comment by ysndr** [comment]
We were discussing this before, the reason behind this is is to check whether we `can_prompt` i.e. we can reach someone on stderr and we can accept their input.
Exactly to allow things like this: `flox xyz | prog` and have normal flox behavior for selecting environments.
In CI we don't have the abilities above, so flox either fails with a message or picks a conflict free env

**Comment by mkenigs** [comment]
@ghudgins @stahnma the current behavior is intentional, can we either close this or confirm we want to change it? Currently we prompt if `stderr.is_tty() && stdin.is_tty()`, but we could change it to `stderr.is_tty() && stdin.is_tty() && stdout.is_tty()`

**Comment by mkenigs** [comment]
Fixed, we can tackle defaults for prompts in another ticket
```
> flox list | xargs
ERROR: can't determine whether to use main at /Users/matthew/flox/main or 1 at /Users/matthew/flox/1
```
